standalone-metastore/metastore-common/src/gen/thrift/gen-javabean/org/apache/hadoop/hive/metastore/api/CompactionType.java

```diff
@@ -12,7 +12,8 @@ public enum CompactionType implements org.apache.thrift.TEnum {
   MINOR(1),
   MAJOR(2),
   REBALANCE(3),
-  ABORT_TXN_CLEANUP(4);
+  ABORT_TXN_CLEANUP(4),
+  SMART(5);
```

Comment:
:), maybe `AUTO`?

Reply:
Renamed to `SMART_OPTIMIZE`

Reply:
Additionally, the compaction type in the HMS is now updated from `SMART_OPTIMIZE` to the actual compaction type (MAJOR or MINOR) that is needed for the table or partition. This update occurs after a `SMART_OPTIMIZE` compaction request is picked up by a compaction worker and compaction evaluator is executed.